File: lib/go/thrift/types/known/timestamp/timestamp.go
package timestamp

import "time"

func Now() *Timestamp {
	return New(time.Now())
}

func New(t time.Time) *Timestamp {
	return &Timestamp{Seconds: int64(t.Unix()), Nanos: int32(t.Nanosecond())}
}

func (t *Timestamp) ToTime() time.Time {
	return time.Unix(t.Seconds, int64(t.Nanos)).UTC()
}

File: lib/rb/lib/thrift/types/known/timestamp/timestamp.rb
require 'thrift/types/known/timestamp/timestamp_types'

module Thrift
  module Types
    module Known
      module Timestamp
        class << self
          def from_time(v)
            Timestamp.from_time(v)
          end

          def now
            from_time(Time.now)
          end
        end

        class Timestamp
          class << self
            def from_time(v)
              Timestamp.new(seconds: v.to_i, nanos: v.nsec)
            end
          end

          def to_time
            Time.at(seconds, nanos, :nsec)
          end
        end
      end
    end
  end
end
